src/gifts.rs

```diff
@@ -75,9 +75,19 @@ pub struct GiftInfo {
 #[apply(apistruct!)]
 pub struct UniqueGiftInfo {
     pub gift: UniqueGift,
-    pub origin: String,
+    pub origin: GiftOrigin,
+    pub last_resale_star_count: Option<u32>,
     pub owned_gift_id: Option<String>,
     pub transfer_star_count: Option<u32>,
+    pub next_transfer_date: Option<u64>,
+}
+
+#[derive(Clone, Copy, Debug, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "snake_case")]
+pub enum GiftOrigin {
```

Comment:
The docs state

> Currently

so its probably a good idea to add `#[non_exhaustive]`

```suggestion
#[derive(Clone, Copy, Debug, Serialize, Deserialize, PartialEq, Eq)]
#[non_exhaustive]
#[serde(rename_all = "snake_case")]
pub enum GiftOrigin {
```